### ISSUE TITLE
Create config map from folder

### ISSUE THREAD

**Issue by viorica-visan1** [terraform-providers/terraform-provider-kubernetes#399]
Hi,

I am not sure how to create a config map with kubernetes_config_map from a folder and I cannot seem to find any example.
It definitely does not work if I just use 
 data {
    config = "${file(/mydir/)"
  }
which I found here https://stackoverflow.com/questions/53667413/is-it-possible-from-file-configmap-using-kubernetes-config-map-resource 
This is also because I did not find any terraform equivalent for file() for directories. 

Is this missing feature ? 
I am able to create the configmap from folder with kubectl. 
I would need this also because I have some binary files in there, that I would not know how to load file by file. 
Could you advice pls ? 

### Terraform Version
Terraform v0.11.13

+ provider.kubernetes v1.5.2

### Affected Resource(s)
kubernetes_config_map

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Expected Behavior
Document the limitations of kubernetes_config_map if they exists OR I can create the resource form the folder.

### Actual Behavior
I cannot create kubernetes_config_map from a folder.

**Comment by pdecat** [comment]
Hi,

creating config maps from folder is indeed not supported by the terraform provider.

What's the result when specifying the files individually?

```
resource "kubernetes_config_map" "test" {
  metadata {
    name = "test"
  }

  data {
   file1 = "${file(/mydir/binary1.data)"
   file2 = "${file(/mydir/binary2.data)"
  }
}
```

Note: it may fail as binary data is supposed to go in [the `binaryData` field of config maps](https://github.com/kubernetes/kubernetes/pull/57938) but the provider does not support that yet.

**Comment by pdecat** [comment]
I've confirmed the issue with binary data that is altered when passed in the `data` field, working on a PR to add support for `binary_data`.

**Comment by viorica-visan1** [comment]
Actually when I try to give the binary files individually, terraform complains:
"<template_file>:1,1-2: Invalid character encoding; All input files must be UTF-8 encoded. Ensure that UTF-8 encoding is selected in your editor., and 1 other diagnostic(s)"

**Comment by pdecat** [comment]
The binary data support is added in https://github.com/terraform-providers/terraform-provider-kubernetes/pull/400

**Comment by viorica-visan1** [comment]
Hi, can I just get a confirmation that the fix in the provider needs terraform 0.12 ?
